pyanaconda/modules/subscription/satellite.py

```diff
@@ -0,0 +1,176 @@
+                          len(provisioning_script))
+                return provisioning_script
+            else:
+                log.debug("subscription: server returned %i code when downloading"
+                          " Satellite provisioning script", result.status_code)
+                result.close()
+                return None
+        except RequestException as e:
+            log.debug("subscription: can't download Satellite provisioning script"
+                      " from %s with proxy: %s. Error: %s", script_url, proxies, e)
+            return None
+
+
+def run_satellite_provisioning_script(provisioning_script=None, run_on_target_system=False):
+    """Run the Satellite provisioning script.
+
+    Each Satellite instance provides a provisioning script that will
+    enable the currently running environment to talk to the given
+    Satellite instance.
+
+    This means that the self-signed certificates of the given
+    Satellite instance will be installed to the system but also some
+    necessary changes will be done to rhsm.conf.
+
+    As we need to provision both the installation *and* target system
+    to talk to Satellite we need to run the provisioning script twice,
+    once in the installation environment and once on the target system.
+
+    This is achieved by running this function first in the installation environment
+    with run_on_target_system == False before a registration attempt.
+    And then in the installation phase with run_on_target_system == True.
+
+    Implementation wise we just always run the script from a tempfile.
+
+    That way we can easily run it in the installation environment as well
+    as in the target system chroot with minimum code needed to make sure
+    it exists where we need it
+
+    :param str provisioning_script: content of the Satellite provisioning script
+                                    or None if no script is available
+    :param str run_on_target_system: run in the target system chroot instead,
+                                     otherwise run in the installation environment
+    :return: True on success, False otherwise
+    :rtype: bool
+    """
+    # first check we actually have the script
+    if provisioning_script is None:
+        log.warning("subscription: satellite provisioning script not available")
+        return False
+
+    # now that we have something to run, check where to run it
+    if run_on_target_system:
+        # run in the target system chroot
+        sysroot = conf.target.system_root
+    else:
+        # run in installation environment
+        sysroot = "/"
+
+    # create the tempfile containing the script in the sysroot in /tmp, just in case
+    sysroot_tmp = util.join_paths(sysroot, "/tmp")
+    with tempfile.NamedTemporaryFile(mode="w+t", dir=sysroot_tmp, prefix="satellite-") as tf:
+        # write the provisioning script to the tempfile
+        tf.write(provisioning_script)
```

Comment:
The file is not flushed or closed when you execute the script. It might be incomplete.

Reply:
Oh, right - we are still inside the context manager and we have to be, so that will not do it for us. :P I think `flush()` should be enough ? I'm a bit afraid if `close()` would perhaps remove the tempfile ? :P

Reply:
`If delete is true (the default), the file is deleted as soon as it is closed.` See: https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile

Reply:
Oops! Then, let's just keep `flush()` but drop `close()`. :)

Temporary files- there could not be anything complicated about those! ;-)